### PR TITLE
koji host and kojira setup

### PR DESCRIPTION
**Pull request by odra**
## Proposed Changes

* chnages koji_host module to include channels to a koji-host;
* `host_arch` became a list of strings and renamed to  `host_archs` to  match koji API;
* adds channel support in koji-builder CR, defaults to "default" and "createrepo";
* adds a new module: koji_user;
* adds a `kojira` user with proper permissions (`repo` perm);
* moves koji-builder and kojira "converge deployment" to koji-hub since both need koji-hub to work.

## Verification Steps

* Run tests `molecule test -s test-local`

### CODE REVIEW

File: mbox-operator/molecule/default/converge/tasks/koji-hub.yml [hunk 1]
```diff
@@ -92,6 +92,41 @@
...
+        kind: '{{ custom_resource_kojibuilder.kind }}'
+        namespace: '{{ namespace }}'
+        name: '{{ custom_resource_kojibuilder.metadata.name }}'
+      register: cr
+      until:
+        - "'Successful' in (cr | json_query('resources[].status.conditions[].reason'))"
+      delay: 10
+      retries: 30
+
+    - name: reate kojia custom resource
```
Comment (Zlopez) [inline mbox-operator/molecule/default/converge/tasks/koji-hub.yml:113]:
```suggestion
    - name: Create kojira custom resource
```

File: mbox-operator/roles/koji-lib/library/koji_host.py [hunk 2]
```diff
@@ -36,10 +36,10 @@
         description:
             - a qualified and unique koji hostname
         required: true
-    arch:
+    archs:
         description:
-            - a valid koji host archtecture
-        default: x86
+            - a list of valid koji host archtectures
```
Comment (Zlopez) [inline mbox-operator/roles/koji-lib/library/koji_host.py:41]:
```suggestion
            - a list of valid koji host architectures
```

**Review by Zlopez — COMMENTED** [review]
Few small typos.

I noticed you removed separate playbook for kojira and koji-builder and added them to koji-hub. Do you think it's a good idea?

**Comment by odra** [comment]
recheck

**Comment by odra** [comment]
I joined both test playbooks into the koji-hub one because they can't be tested on their own anymore, both need a running koji-hub instance.

**Comment by morucci** [comment]
recheck

File: mbox-operator/molecule/test-local/destroy.yml [hunk 3]
```diff
@@ -29,7 +29,6 @@
       until: docker_jobs.finished
       retries: 300
       with_items: "{{ server.results }}"
-      no_log: item.failed
```
Comment (odra) [inline mbox-operator/molecule/test-local/destroy.yml:32]: this was causing the tests to fail so I removed it.

**Review by Zlopez — APPROVED** [review]

**Pull request merged**